pyccel/parser/semantic.py

```diff
@@ -823,6 +826,19 @@ def _handle_function(self, expr, func, args, **settings):
                         severity = 'fatal')
 
             args, kwargs = split_positional_keyword_arguments(*args)
+            if func in (CupyArray, CudaArray, NumpyArray):
+                if 'device' in self.scope.decorators or 'kernel' in self.scope.decorators:
```

Comment:
I don't see these decorators in the `decorators.py` file. Are these cupy objects?

Reply:
It's odd, i think it should be added to `decorators.py`.

Reply:
Hmmm, ok. I also realised the `kernel` decorator is there, it's just out of order in the `__all__` variable in decorators.py (the rest are alphabetical)